service/keyv2/key.go

```diff
@@ -23,43 +23,43 @@ const (
 	ProfileNameTemplate = "EC2-K8S-Role"
 )
 
-func AutoScalingGroupName(customObject awstpr.CustomObject, groupName string) string {
+func AutoScalingGroupName(customObject v1alpha1.AWSConfig, groupName string) string {
 	return fmt.Sprintf("%s-%s", ClusterID(customObject), groupName)
 }
 
-func AvailabilityZone(customObject awstpr.CustomObject) string {
+func AvailabilityZone(customObject v1alpha1.AWSConfig) string {
 	return customObject.Spec.AWS.AZ
 }
 
-func BucketName(customObject awstpr.CustomObject, accountID string) string {
+func BucketName(customObject v1alpha1.AWSConfig, accountID string) string {
 	return fmt.Sprintf("%s-g8s-%s", accountID, ClusterID(customObject))
 }
 
-func BucketObjectName(customObject awstpr.CustomObject, prefix string) string {
+func BucketObjectName(customObject v1alpha1.AWSConfig, prefix string) string {
```

Comment:
We should stop using the cluster version here. Instead we can provide the k8scloudconfig version that has been selected by the operator. But I want to do that as a separate change.

Reply:
> we can provide the k8scloudconfig version that has been selected by the operator

I don't follow. How does the operator select this? When we work on the cloudconfig versioning lets keep in mind that we are going to have a separate operator and we read the cloudconfig/ignitionmagic like we do with the secrets.

Reply:
We were selecting the cloudconfig based on the cluster version and defaulting it when it doesn't exist. That doesn't work as we want to remove the cluster version. 

https://github.com/giantswarm/aws-operator/blob/2683d10cb5bdd3a307326984f280908642e33c63/service/cloudconfigv1/cloud_config.go#L56

So the cloudconfigv2 resource should use `V2_0_0` since it has the `apiextensions` support. Is that correct?

https://github.com/giantswarm/k8scloudconfig/blob/4b0542adc876f2e34129d95bc06e5557415c320e/template.go#L27

This is until we have the operator. I agree it can then deal with generating the correct cloudconfig/ignition.  We can update the cloudconfig service to just get the output that has been generated like we do with the certs secrets.

Reply:
Yes, in AWS we should use `V_2_0_0`. In KVM we have to use `V_1_1_0` which is the same then as in AWS but without the experimental encryption support, which as to get fixed, so we can also go with `V_2_0_0` in KVM.

Reply:
Ack. Thanks, I'll do that in the next PR.